EstateAdministrationUI/Areas/Estate/Controllers/ContractController.cs

```diff
@@ -0,0 +1,159 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace EstateAdministrationUI.Areas.Estate.Controllers
+{
+    using System.Diagnostics.CodeAnalysis;
+    using System.Linq.Expressions;
+    using System.Security.Claims;
+    using System.Threading;
+    using BusinessLogic.Models;
+    using Common;
+    using Factories;
+    using Microsoft.AspNetCore.Authentication;
+    using Microsoft.AspNetCore.Authorization;
+    using Microsoft.AspNetCore.Mvc;
+    using Models;
+    using Newtonsoft.Json;
+    using Services;
+    using Shared.Logger;
+
+    [ExcludeFromCodeCoverage]
+    [Authorize]
+    [Area("Estate")]
+    public class ContractController : Controller
+    {
+        #region Fields
+
+        /// <summary>
+        /// The API client
+        /// </summary>
+        private readonly IApiClient ApiClient;
+
+        /// <summary>
+        /// The view model factory
+        /// </summary>
+        private readonly IViewModelFactory ViewModelFactory;
+
+        #endregion
+
+        #region Constructors
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="ContractController" /> class.
+        /// </summary>
+        /// <param name="apiClient">The API client.</param>
+        /// <param name="viewModelFactory">The view model factory.</param>
+        public ContractController(IApiClient apiClient,
+                                  IViewModelFactory viewModelFactory)
+        {
+            this.ApiClient = apiClient;
+            this.ViewModelFactory = viewModelFactory;
+        }
+
+        [HttpGet]
+        public async Task<IActionResult> CreateContract(CancellationToken cancellationToken)
+        {
+            CreateContractViewModel viewModel = new CreateContractViewModel();
+
+            return this.View("CreateContract", viewModel);
+        }
+
+        /// <summary>
+        /// Validates the model.
+        /// </summary>
+        /// <param name="viewModel">The view model.</param>
+        /// <returns></returns>
+        private Boolean ValidateModel(CreateContractViewModel viewModel)
+        {
+            return this.ModelState.IsValid;
+        }
+
+        [HttpPost]
+        public async Task<IActionResult> CreateContract(CreateContractViewModel viewModel, CancellationToken cancellationToken)
+        {
+            // Validate the model
+            if (this.ValidateModel(viewModel))
+            {
+                String accessToken = await this.HttpContext.GetTokenAsync("access_token");
+
+                CreateContractModel createContractModel = this.ViewModelFactory.ConvertFrom(viewModel);
+
+                // All good with model, call the client to create the golf club
+                CreateContractResponseModel createContractResponseModel =
```

Comment:
Codacy found an issue: [Remove the unused local variable 'createContractResponseModel'.](https://app.codacy.com/manual/stuart_ferguson1/EstateAdministrationUI/pullRequest?prid=6070534)